src/query/sortResults.ts

```diff
@@ -0,0 +1,91 @@
+import debug from 'debug'
+import get from 'lodash.get'
+import { InternalEntity } from 'src/glossary'
+import { OrderBy, SortDirection } from './queryTypes'
+
+const log = debug('sortResults')
+
+type FlatSortCriteria = [string[], SortDirection]
+
+export function warnOnIneffectiveSortingKeys(obj: Record<string, any>): void {
+  const [mainCriteria, ...siblings] = Object.keys(obj)
+
+  if (siblings.length > 0) {
+    console.warn(
+      'Sorting by "%s" has no effect: already sorted by "%s".',
+      siblings.join(','),
+      mainCriteria,
+    )
+  }
+}
+
+function flattenSortCriteria(
+  orderBy: OrderBy<any>[],
```

Comment:
This 'any' is hiding some sneaky type stuff going on.

Really this function ought to be generically typed and then we can check against that generic type rather than need to use any:

```ts
function flattenSortCriteria<Entity>(
  orderBy: OrderBy<Entity>[],
```

Then on lines 33-35:
```ts
    const property = Object.keys(properties)[0] as keyof OrderBy<Entity>
    const sortDirection = properties[property]!
    const path = propertyPath.concat(property.toString())
```